docs/docker.rst

```diff
@@ -0,0 +1,162 @@
+
+Now you are able to use all regular :doc:`Rally commands <command_line_reference/>`, bearing in mind the aforementioned :ref:`limitations <docker_limitations>`.
+
+Configuration
+-------------
+
+The Docker image ships with a default configuration file under ``/rally/.rally/rally.ini``.
+To customize Rally you can create your own ``rally.ini`` and bind mount it using::
+
+    docker run -v /home/<myuser>/custom_rally.ini:/rally/.rally/rally.ini elastic/rally ...
+
+Persistence
+-----------
+
+It is highly recommended to use a local bind mount (or a `named volume <https://success.docker.com/article/different-types-of-volumes>`_) for the directory ``/rally/.rally`` in the container.
+This will ensure you have persistence across invocations and any tracks downloaded and extracted can be reused, reducing the startup time.
+You need to ensure the UID is ``1000`` (or GID is ``0`` especially in OpenShift) so that Rally can write to the bind-mounted directory.
+
+If your local bind mount doesn't contain a ``rally.ini`` the container will create one for you during the first run.
+
+Example::
+
+    mkdir myrally
+    sudo chgrp 0 myrally
+
+    # First run will generate the rally.ini
+    docker run --rm -v $PWD/tmpdir:/rally/.rally elastic/rally --track=nyc_taxis --test-mode --pipeline=benchmark-only --target-hosts=es01:9200
+
+        ____        ____
+       / __ \____ _/ / /_  __
+      / /_/ / __ `/ / / / / /
+     / _, _/ /_/ / / / /_/ /
+    /_/ |_|\__,_/_/_/\__, /
+                    /____/
+
+    Running simple configuration. Run the advanced configuration with:
+
+      esrally configure --advanced-config
+
+    * Setting up benchmark root directory in /rally/.rally/benchmarks
+    * Setting up benchmark source directory in /rally/.rally/benchmarks/src/elasticsearch
+
+    Configuration successfully written to /rally/.rally/rally.ini. Happy benchmarking!
+
+    More info about Rally:
+
+    * Type esrally --help
+    * Read the documentation at https://esrally.readthedocs.io/en/latest/
+    * Ask a question on the forum at https://discuss.elastic.co/c/elasticsearch/rally
+
+    # now run our benchmark
+    docker run --rm -v $PWD/tmpdir:/rally/.rally elastic/rally --track=nyc_taxis --test-mode --pipeline=benchmark-only --target-hosts=es01:9200
+
+    ...
+
+    # inspect results
+    $ tree myrally/benchmarks/races/
+    myrally/benchmarks/races/
+    └── 2019-06-05-14-03-44
+        └── race.json
+
+    1 directory, 1 file
+
+
+In case you forgot to bind mount a directory, the Rally Docker image will create an `anonymous volume <https://success.docker.com/article/different-types-of-volumes>`_ for ``/rally/.rally`` to ensure logs and results get persisted even after the container has terminated.
+
+For example, after executing our earlier quickstart example ``docker run elastic/rally --track=nyc_taxis --test-mode --pipeline=benchmark-only --target-hosts=es01:9200``, ``docker volume ls`` shows a volume:::
```

Comment:
Nit: This line ends with three colons but it only requires two.

Reply:
Addressed in https://github.com/elastic/rally/pull/702/commits/af1a12402329af7bf27343bc33d63d3fef7abd4b

Reply:
Addressed in https://github.com/elastic/rally/pull/702/commits/af1a12402329af7bf27343bc33d63d3fef7abd4b